### DISCUSSION TITLE
Open "git add -p" window in new tab rather than horizontal split

### DISCUSSION THREAD

**Discussion by Hubro** [tpope/vim-fugitive#1830] [Q&A] [answered]
If I open fugitive `:G` then press `P` on a file to do `git add -p`, the interactive window pops up as another horizontal split. This means that most of the time, the current hunk is bigger than the window, so I have to scroll.

Is it possible to configure fugitive to open this window in a new tab or vertical split (covering the entire height of the window) instead?

(If using a Neovim floating window is also possible, that would be even better)

**Comment by tpope** [comment, marked as answer]
A new tab is possible with `:tab Git add --patch`. The map isn't configurable (Edit: though it can be remapped like any other map), but I suppose I'll try changing the default.  There is already precedent for `cvc` forcing a new tab.

Neovim floating windows, as I understand them, are not a drop in replacement for regular windows in the same way that tabs are.

**Reply by Hubro** [reply]
Cool, thanks for the answer! Yeah, I think a tab is probably a more sane default than a second horizontal split.

In any case, I just realized that I can stage/unstage chunks directly from the `:G` screen, so I probably won't be using the add patch screen anyway. I keep getting surprised by how awesome this plugin is :smile:

> Neovim floating windows, as I understand them, are not a drop in replacement for regular windows in the same way that tabs are.

True, but it would still be awesome :grin: Worth thinking about, since Neovim supports floating windows and Vim now supports "popups", which by the looks of it is basically the same thing.

**Comment by rouflak** [comment]
@tpope Since a recent update the interactive add of fugitive opens in a new tab as a default behavior.
Is there a configuration that could help to meet the previous behavior?

Thx for helping

**Reply by tpope** [reply]
Define whatever map you want in `after/ftplugin/fugitive.vim`.
